app/views/tasks/index.html.erb

```diff
@@ -0,0 +1,28 @@
+<header>
+<h1>Task List</h1>
+<button>
+ <%= link_to "Add Task", new_task_path %></button>
+
+<ul class="task-list">
+  <% @tasks.each do |task| %>
+    <li>
+      <strong>Task:</strong> <%= link_to "#{task.name}", task_path(task)%>
+      <strong>Description:</strong> <%="#{task.description}"%>
+      <strong>Date:</strong> <%= "#{task.date}" %> 
+
+      <div><br>
+      <%= link_to  "Edit Task | ", edit_task_path(task) %>
+      <%= link_to  "Delete Task | ", task_path(task), method: :delete, data: {confirm: "Are you sure you want to delete?"}%><br>
+      <%= link_to "Task completed | ", completed_path(task), method: :patch %>
```

Comment:
There is a small bug here. Both the "Task completed" and the "Task Not Completed" link toggle the completed setting to the opposite setting that it was currently. These buttons need to have a reliable behavior. One way to achieve this, would be to have 2 distinct controller actions: one `mark_incomplete` the other `mark_complete`.